pkg/nsx/services/vpc/vpc.go

```diff
@@ -549,21 +550,31 @@ func (s *VPCService) CreateorUpdateVPC(obj *v1alpha1.VPC) (*model.Vpc, *common.V
 	return &newVpc, &nc, nil
 }
 
-func (s *VPCService) Cleanup() error {
+func (s *VPCService) Cleanup(ctx context.Context) error {
 	vpcs := s.ListVPC()
 	log.Info("cleaning up vpcs", "Count", len(vpcs))
 	for _, vpc := range vpcs {
-		if err := s.DeleteVPC(*vpc.Path); err != nil {
-			return err
+		select {
+		case <-ctx.Done():
+			return util2.TimeoutFailed
+		default:
+			if err := s.DeleteVPC(*vpc.Path); err != nil {
+				return err
+			}
 		}
 	}
 
 	ipblocks := s.IpblockStore.List()
 	log.Info("cleaning up ipblocks", "Count", len(ipblocks))
 	for _, ipblock := range ipblocks {
 		ipb := ipblock.(*model.IpAddressBlock)
-		if err := s.deleteIPBlock(*ipb.Path); err != nil {
-			return err
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("failed to clean up ippool, context cancelled")
```

Comment:
nit: Any reason this one doesn't return TimeoutFailed?

Reply:
fixed